mgmt/imgmgr/src/imgmgr.c

```diff
@@ -301,6 +301,111 @@ imgr_read_info(int image_slot, struct image_version *ver, uint8_t *hash,
     return rc;
 }
 
+/*
+ * Read the current running image's build hash
+ *
+ * @param hash Ptr to hash to be filled up
+ * @param hashlen Length of hash to return
+ *
+ * Returns -2 if either of the argument is 0 or NULL
+ * Returns -1 if area is not readable
+ * Returns 0 if image in slot is ok
+ * Returns 1 if there is not a full image
+ * Returns 2 if slot is empty
+ */
+int
+imgr_get_current_hash(uint8_t *hash, uint16_t hashlen)
```

Comment:
Can this function (`imgr_get_current_hash()`) just call `imgr_read_info()` rather than duplicate all its code?

Reply:
Ok, that seems a fine enough ask, I actually wanted to just read the exact number of bytes and not read more bytes from the flash than needed but I guess reading does not take that many cycles, so I will just reuse the code we already have. Thanks.